### ISSUE TITLE
orderby operators missing from api surface

### ISSUE THREAD

**Issue by patsissons** [ReactiveX/IxJS#30]
**IxJS version:** 2.0.1

**Code to reproduce:** `Iterable.from([1]).orderBy(x => x)`

**Expected behavior:** returns an ordered iterable

**Actual behavior:** `property 'orderBy' does not exist on type 'Iterable<T>'`

**Additional information:**

The operator add `import` is just missing from the api surface, not sure if that was intentional or not.

**Comment by patsissons** [comment]
fairly low priority as we can easily add the operators as needed.

**Comment by ichpuchtli** [comment]
@patsissons  Is ordering a responsibility of an iterable?. Only once its converted to a finite array can you begin to order it no?

**Comment by ichpuchtli** [comment]
@patsissons never mind I see now that matt is collection/committing iterables to finite arrays in ix, for some reason I thought this project was being limited to operators you can perform without collection.

**Comment by jonatannyberg** [comment]
@patsissons This seems to be the case in 2.1.0 as well, for `orderBy` and `groupBy`. And no we can not easily add the operators as needed. Because the typings are wrong. 

Example to reproduce in typescript (2.5.2): 
```
import { Iterable } from 'ix';
import 'ix/add/iterable-operators/groupby';

myIterable.groupBy(item => item.prop);
```
The above code can not find `groupBy`, it is missing from 'Ix.d.ts'. If the `import` is changed to 'ix/types/add/iterable-operators/groupby' it would compile. But that would not work at runtime. 

The solution would be to import the missing operators in Ix.d.ts, like it is done for the rest of the operators.

**Comment by jonatannyberg** [comment]
After trying to use 2.1.0 for a while with typescript now this issue seems to apply to several operators. Rendering this virtually unusable to any typescript development. I might be missing something here but I can not seem to figure out how to get this to work at all with typescript in its current state.

**Comment by trxcllnt** [comment]
@jonatannyberg @patsissons it seems like VSCode is having an issue resolving the types when they're separated from the transpiled files, but I'll review the options when I get a chance this weekend. In the meantime, I've got VSCode working via paths entry in tsconfig:
```json
{
    "compilerOptions": {
        "baseUrl": ".",
        "paths": {
            "ix/*": ["node_modules/ix/types/*"]
        }
    }
}
```

I realize it's not ideal, but I hope it helps in the short term.

**Comment by jonatannyberg** [comment]
@patsissons @trxcllnt 
Might be hard to get this to work without a shared tree for transpired files and the types. 

I looked in to how RxJS solved it and there the transpiled files and the types share the same tree. 

One solution that might work is to simply import the missing operators in Ix.ts; that solves the issue when importing from `ix`. It will still cause problems when trying to import directly from iterator however, for example: 
```
import { IterableX as Iterable } from 'ix/iterable';
import 'ix/add/iterable/of';
import 'ix/add/iterable-operators/map';
``` 
Can this ever work without having transpiled files and types share the same tree or modifying paths in tsconfig?

**Comment by patsissons** [comment]
I apologize for not responding with my current working solution earlier.

`tsconfig.json` (partial)
```json
{
  "compilerOptions": {
    "paths": {
      "ix/*": [ "node_modules/ix/types/*" ]
    }
  }
}
```

`webpack.config.js` (partial)
```js
module.exports = {
  resolve: {
    alias: {
      // use es5 target for ix modules
      'ix': 'ix/targets/es5/cjs',
      // this ensures we pull in the es5 ix api surface
      'ix$': 'ix/Ix',
    },
  },
};
```

**Comment by jonatannyberg** [comment]
@patsissons it is posible to get this to work with webpack/systemjs or similar by telling the runtime how to locate the files and by telling the transpiler where to find the .d.ts files at compile time. The core issue still remains however. 

When compiling for a pure node run time (not building with webpack) it is very hard to get this to work. It is possible to get it to compile with the path workaround mentioned above, but this results in node not finding needed files at runtime. The only way I can work out to get around this is to compile the source myself, and this is not ideal. 

Any work being done to resolve this issue?

**Comment by trxcllnt** [comment]
@jonatannyberg yes, I am working on a PR now to address this

**Comment by trxcllnt** [comment]
@jonatannyberg @patsissons @ichpuchtli we just published #45 as `ix@2.1.2` on npm, which should address these issues. We now export the types for the Grouped and Ordered internals, and all the files in the module directly correspond to the path structure in the docs. You should be able to remove any path mappings you have after the upgrade. I'll leave this issue open for a few days so you can still use this thread if there's any trouble. Otherwise, happy hacking!

**Comment by patsissons** [comment]
checks out. removed my mappings in tsconfig and webpack config and everything is connecting together nicely. importing additional operators through `'ix/add/*'` still works properly (without the mappings). Nice work!

**Comment by trxcllnt** [comment]
@patsissons 🎉 🙏

**Comment by DanHarman** [comment]
@trxcllnt I must be doing something wrong as I'm still having to manually import orderby for it to appear?

I'm using ix@2.1.4. with the following tsconfig:

```
{
    "compilerOptions": {
        "allowSyntheticDefaultImports": true,
        "baseUrl": ".",
        "module": "es2015",
        "moduleResolution": "node",
        "target": "es6",
        "jsx": "react",
        "experimentalDecorators": true,
        "sourceMap": true,
        "skipDefaultLibCheck": true,
        "strict": true,
        "lib": [
            "es6",
            "es2017",
            "dom",
            "esnext.asynciterable" /* <-- For ix */
        ],
        "types": ["webpack-env"],
        "typeRoots": [
            "../@types"
        ]
    },
    "exclude": [
        "bin",
        "node_modules"
    ]
}
```

Any suggestions appreciated!
